src/test/groovy/org/mockito/release/notes/vcs/RevisionDateProviderTest.groovy

```diff
@@ -12,9 +12,21 @@ class RevisionDateProviderTest extends Specification {
     def "provides revision dates"() {
         runner.run("git", "log", "--pretty=%ad", "--date=iso", "v1.0.0", "-n", "1") >> "\n2017-01-29 08:14:09 -0800\n"
         runner.run("git", "log", "--pretty=%ad", "--date=iso", "v2.0.0", "-n", "1") >> "\n2017-01-30 10:14:09 -0400\n"
+        runner.run("git", "log", "--pretty=%ad", "--date=iso", "v3.0.0", "-n", "1") >> "2017-04-11 13:59:59 +0000"
 
         expect:
         DateFormat.formatDate(provider.getDate("v1.0.0")) == "2017-01-29"
         DateFormat.formatDate(provider.getDate("v2.0.0")) == "2017-01-30"
+        DateFormat.formatDate(provider.getDate("v3.0.0")) == "2017-04-11"
+    }
+
+    def "fails if revision number is incorrect what causes git call to fail"() {
+        runner.run("git", "log", "--pretty=%ad", "--date=iso", "v1.0.0", "-n", "1") >> "fatal: ambiguous argument 'v1.0.0'"
+
+        when:
+        provider.getDate("v1.0.0")
+
+        then:
+        thrown IllegalArgumentException
```

Comment:
For extra safety, let's add a simple assertion message on the exception content. We can assert on full exception message or part of it, whatever is simpler.

Otherwise, we might not be testing the right thing. IllegalArgumentException is a generic exception that can be thrown for other reasons than the one that we expect in this test.